lingua_franca/lang/format_fa.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from lingua_franca.lang.format_common import convert_to_mixed_fraction
+from lingua_franca.lang.common_data_fa import \
+    _farsiOnes, _farsiTens, _farsiHundreds, _farsiBig, _farsiSeperator, \
+    _farsiFrac, _farsiFracBig, _FRACTION_STRING_FA
+import math
+from lingua_franca.internal import lookup_variant
+from enum import IntEnum
+
+class NumberVariantFA(IntEnum):
+    CONVERSATIONAL = 0
+    FORMAL = 1
+
+lookup_number = lookup_variant({
+    "default": NumberVariantFA.CONVERSATIONAL,
+    "conversational": NumberVariantFA.CONVERSATIONAL,
+    "formal": NumberVariantFA.FORMAL,
+})
+
+def _applyVariant(text, variant):
+    if variant == NumberVariantFA.FORMAL:
+        for key, value in _formal_variant.items():
+            text = text.replace(value, key)
+    return text
+
+def _handleVariant(func):
+    @lookup_variant({
+        "default": NumberVariantFA.CONVERSATIONAL,
+        "conversational": NumberVariantFA.CONVERSATIONAL,
+        "formal": NumberVariantFA.FORMAL,
+    })
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+        print(kwargs, result)
```

Comment:
@JarbasAl @ChanceNCounter I tried to handle variant formats with a local decorator, but strangely, kwargs is empty always! (so tests are failing). I did not understand where the problem came from. Can you take a look?

Reply:
see my other comment, you are not using the lookup_variant decorator like i intended it to be used, the logic to handle the variant should be inside the the nice_number method

```python
if variant == NumberVariantFA.FORMAL:
    # changes required for this variant
elif variant == NumberVariantFA.CONVERSATIONAL:
    # changes required for this variant
```